src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const cloneArray = [];
+  const cloneObject = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(cloneObject, action.extraData);
+        cloneArray.push({ ...cloneObject });
```

Comment:
You do  `cloneArray.push({ ...cloneObject });` in any case, so move it to the end of the logic and write only once